lib/src/sql/array.rs

```diff
@@ -317,13 +318,18 @@ pub trait Uniq<T> {
 
 impl Uniq<Array> for Array {
 	fn uniq(mut self) -> Array {
-		for x in (0..self.len()).rev() {
-			for y in (x + 1..self.len()).rev() {
-				if self[x] == self[y] {
-					self.remove(y);
-				}
+		let mut set: HashSet<&Value> = HashSet::new();
+		let mut to_remove: Vec<usize> = Vec::new();
+		for i in 0..self.len() {
+			if set.contains(&self[i]) {
+				to_remove.push(i);
```

Comment:
Seems like `to_remove` may contain duplicates, which will cause the wrong items to be removed later on.

Reply:
it shouldn't, as I'm not actually removing anything in this iteration.

Reply:
Ah yes, never mind. Each `i` will be distinct, even if they refer to the same duplicate item.

Reply:
It's just that, in the worst case (array of **n** equal items), the time complexity will be still be **O(n<sup>2</sup>)**

Using [Abolish](https://github.com/surrealdb/surrealdb/blob/d5a378b1beae5d0ecee01e74bd89e15921cc5cb6/lib/src/sql/array.rs#L204-L224) where the closure does a hash lookup would probably preserve order and be faster in the worst case due to the inner workings of `retain`.